GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java

```diff
@@ -196,6 +208,29 @@ public void onServiceConnected(ComponentName name, IBinder service) {
             //get the gpsDataCaptureService
             gpsDataCaptureService =
                     ((GpsDataCaptureService.GpsDataCaptureBinder) service).getService();
+
+            //start capture via intent
+            if(adbIntent != null) {
+                //Extra the location api type
+                if(adbIntent.getExtras() != null){
+                    Log.d(TAG, "Intent extra is " + adbIntent.getExtras().toString());
+                    boolean type_from_intent = adbIntent.getBooleanExtra("fused_location_type", false);
+                    Log.d(TAG, "fused_location_type: " + type_from_intent);
+                    if(type_from_intent){
+                        locationApiType = LocationApiType.FUSEDLOCATIONPROVIDERCLIENT;
+                    }
+                    Log.d(TAG, "LocationApiType: " + locationApiType);
+                }
+                //Start capture if the action is START_CAPTURE
+                if (adbIntent.getAction() != null) {
```

Comment:
You may want to compare it with action string "com.google.gpsdatacapturer.START_CAPTURE"

Reply:
Yes, I did it on line 228.